test/fx/test_gradual_type.py

```diff
@@ -235,6 +236,100 @@ def forward(self, x: TensorType((1, 2, 3, 5))):
+        graph = ast_rewriter.trace(B)
+        traced = GraphModule(ast_rewriter.root, graph, "gm")
+        tc = GraphTypeChecker({}, traced)
+        tc.type_check()
+
+        for n in graph.nodes:
+            if n.op == 'placeholder':
+                assert n.type == TensorType((2, 2, 5, 4))
+            if n.op == 'output':
+                assert n.type == TensorType((2, 2, 5, 4))
+            if n.op == 'call_module':
+                assert n.type == TensorType((2, 2, 5, 4))
+            if n.op == 'call_function':
+                assert n.type == TensorType((2, 2, 5, 4))
+
+    def test_type_check_batch_norm_2D_false(self):
+        class BasicBlock(torch.nn.Module):
+
+            def __init__(self, inplanes, planes, norm_layer=None):
+                super(BasicBlock, self).__init__()
+                if norm_layer is None:
+                    norm_layer = torch.nn.BatchNorm2d
+                self.bn1 = norm_layer(planes)
+
+            def forward(self, x: TensorType((2, 2, 5))):
+                identity = x
+                out: TensorType((2, 2, Dyn, 4)) = self.bn1(x)
+                out += identity
+                return out
+
+        B = BasicBlock(2, 2)
+        ast_rewriter = RewritingTracer()
+        graph = ast_rewriter.trace(B)
+        traced = GraphModule(ast_rewriter.root, graph, "gm")
+        tc = GraphTypeChecker({}, traced)
+        with self.assertRaises(TypeError):
+            tc.type_check()
+
+    def test_type_check_batch_norm_2D_broadcast(self):
+        class BasicBlock(torch.nn.Module):
+
+            def __init__(self, inplanes, planes, norm_layer=None):
+                super(BasicBlock, self).__init__()
+                if norm_layer is None:
+                    norm_layer = torch.nn.BatchNorm2d
+                self.bn1 = norm_layer(planes)
+
+            def forward(self, x: Dyn):
+                identity = x
+                out: TensorType((2, 2, Dyn, 4)) = self.bn1(x)
+                out += identity
+                return out
+
+        B = BasicBlock(2, 2)
+        ast_rewriter = RewritingTracer()
+        graph = ast_rewriter.trace(B)
+        traced = GraphModule(ast_rewriter.root, graph, "gm")
+        tc = GraphTypeChecker({}, traced)
+        tc.type_check()
+        for n in graph.nodes:
+            if n.op == 'placeholder':
+                assert n.type == TensorType((Dyn, Dyn, Dyn, Dyn))
+            if n.op == 'call_function':
+                assert n.type == TensorType((Dyn, Dyn, Dyn, Dyn))
+            if n.op == 'output':
+                assert n.type == TensorType((Dyn, Dyn, Dyn, Dyn))
+            if n.op == 'call_module':
+                assert n.type == TensorType((2, 2, Dyn, 4))
+
+        B = BasicBlock(1, 1)
+        ast_rewriter = RewritingTracer()
+        graph = ast_rewriter.trace(B)
+        traced = GraphModule(ast_rewriter.root, graph, "gm")
+        tc = GraphTypeChecker({}, traced)
+        with self.assertRaises(TypeError):
+            tc.type_check()
```

Comment:
When I look at the error message, looks like we lost `_DynType.__repr__`:

```
TypeError: Cannot apply BatchNorm2d(1, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True) with input type TensorType[(<torch.fx.tensor_type._DynType object at 0x7f5537046940>, <torch.fx.tensor_type._DynType object at 0x7f5537046940>, <torch.fx.tensor_type._DynType object at 0x7f5537046940>, <torch.fx.tensor_type._DynType object at 0x7f5537046940>)] and existing type TensorType[(2, 2, <torch.fx.tensor_type._DynType object at 0x7f5537046940>, 4)] on bn1
```

Can you add that?